tests/utils/test_progress.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import time
+from unittest.mock import Mock, patch
+
+import pytest
+
+from utils.progress import Progress
+
+
+class TestProgressDebouncing:
+    """Test Progress class debouncing functionality."""
+
+    @patch('utils.progress.displayer')
+    def test_progress_basic_functionality(self, mock_displayer):
+        """Test that basic progress functionality still works."""
+        mock_display = Mock()
+        mock_displayer.return_value = mock_display
+
+        # Create progress bar
+        with Progress(total=5, description="Test", min_interval_sec=0.1) as pbar:
+            for _i in range(5):
+                pbar.update(1)
+
+        # Should have made display calls
+        assert mock_display.call_count >= 1
+
+    @patch('utils.progress.displayer')
+    def test_progress_rapid_updates_fallback(self, mock_displayer):
+        """Test rapid updates fall back to manual rate limiting when no event loop."""
+        mock_display = Mock()
+        mock_displayer.return_value = mock_display
+
+        # Create progress bar with short interval
+        with Progress(total=10, description="Rapid", min_interval_sec=0.05) as pbar:
+            time.monotonic()
+
+            # Rapid updates (no sleep)
+            for _i in range(10):
+                pbar.update(1)
+
+            time.monotonic()
+
+        # Should have limited display calls due to rate limiting
+        # (Exact count depends on timing, but should be less than 10)
+        assert mock_display.call_count < 10
+        assert mock_display.call_count >= 2  # At least initial + final
+
+    @patch('utils.progress.displayer')
+    def test_progress_forced_updates_bypass_debouncing(self, mock_displayer):
+        """Test that forced updates bypass debouncing."""
+        mock_display = Mock()
+        mock_displayer.return_value = mock_display
+
+        pbar = Progress(total=3, description="Forced", min_interval_sec=0.1)
+
+        # Multiple forced updates should all go through
+        initial_count = mock_display.call_count
+        pbar._display(force=True)
+        pbar._display(force=True)
+        pbar._display(force=True)
```

Comment:
If you agree that we can remove the `force` parameter, perhaps this test is no longer needed?